### PR TITLE
Added ability to support validation of dictionaries

### PR DESCRIPTION
**Pull request by nkimaina**

### CODE REVIEW

**Comment by nkimaina** [comment]
This is to allow repetition of elements with same structure whose keys are defined. e.g all configurations that will go under widgetConfig will have some common properties like 'name', 'config' etc

File: src/module-config/module-config.ts [hunk 1]
```diff
@@ -140,6 +144,16 @@ const validateConfig = (schema, config, keyPath = "") => {
   }
 };
 
+function validateDictionary(dictionarySchema, config, keyPath) {
```
Comment (FlorianRappl) [inline src/module-config/module-config.ts:147]: Can we type these parameters?

File: src/module-config/module-config.ts [hunk 2]
```diff
@@ -122,6 +122,10 @@ function mergeConfigs(configs: Config[]) {
 // of the provided properties exist in the schema. Run validators
 // where present in the schema.
 const validateConfig = (schema, config, keyPath = "") => {
```
Comment (FlorianRappl) [inline src/module-config/module-config.ts:124]: Missing types of parameters

File: src/module-config/module-config.ts [hunk 3]
```diff
@@ -140,6 +144,16 @@ const validateConfig = (schema, config, keyPath = "") => {
   }
 };
 
+function validateDictionary(dictionarySchema, config, keyPath) {
+  for (let [key, value] of Object.entries(config)) {
+    validateConfig(
+      dictionarySchema.dictionaryElements,
+      value,
+      keyPath + "." + key
```
Comment (FlorianRappl) [inline src/module-config/module-config.ts:152]: I would prefer a template string

```ts
`${keyPath}.${key}`
```

**Comment by FlorianRappl** [comment]
(Just for completeness:

You don't need to type the parameters with default values, unless you want / need to be overly specific:

```ts
keyPath: string = ""
```

could be

```ts
keyPath = ""
```

Reason: TS will infer that this is of type `string`.)

**Pull request merged**